src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets

```diff
@@ -366,14 +405,44 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
+  <Target Name="ResolveRazorComponentInputs">
+    <!--
+    Gathers input source files for Razor component generation. This is a separate target so that we can avoid
+    lots of work when there are no inputs for code generation.
+
+    NOTE: This target is called as part of an incremental build scenario in VS. Do not perform any work
+    outside of calculating RazorComponent items in this target.
+    -->
+
+    <ItemGroup Condition="'$(EnableDefaultRazorComponentItems)'=='true'">
+      <RazorComponent Include="@(Content)" Condition="'%(Content.Extension)'=='.razor'" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="AssignRazorComponentTargetPaths" Condition="'@(RazorComponent)' != ''">
+    <AssignTargetPath Files="@(RazorComponent)" RootFolder="$(MSBuildProjectDirectory)">
+      <Output TaskParameter="AssignedFiles" ItemName="RazorComponentWithTargetPath" />
+    </AssignTargetPath>
+
+    <ItemGroup>
+      <RazorComponentWithTargetPath Condition="'%(RazorComponentWithTargetPath.GeneratedOutput)' == ''">
+        <GeneratedOutput>$(RazorComponentIntermediateOutputPath)%(RazorComponentWithTargetPath.TargetPath)$(RazorGenerateOutputFileExtension)</GeneratedOutput>
+      </RazorComponentWithTargetPath>
+
+      <RazorComponentWithTargetPath Condition="'%(RazorComponentWithTargetPath.GeneratedDeclaration)' == ''">
+        <GeneratedDeclaration>$(RazorComponentIntermediateOutputPath)%(RazorComponentWithTargetPath.TargetPath)$(RazorGenerateOutputDefinitionFileExtension)</GeneratedDeclaration>
+      </RazorComponentWithTargetPath>
+    </ItemGroup>
+  </Target>
+
   <Target Name="AssignRazorGenerateTargetPaths" Condition="'@(RazorGenerate)' != ''">
     <AssignTargetPath Files="@(RazorGenerate)" RootFolder="$(MSBuildProjectDirectory)">
       <Output TaskParameter="AssignedFiles" ItemName="RazorGenerateWithTargetPath" />
     </AssignTargetPath>
 
     <ItemGroup>
       <RazorGenerateWithTargetPath Condition="'%(RazorGenerateWithTargetPath.GeneratedOutput)' == ''">
-        <GeneratedOutput>$(RazorGenerateIntermediateOutputPath)$([System.IO.Path]::ChangeExtension('%(RazorGenerateWithTargetPath.TargetPath)', '$(RazorGenerateOutputFileExtension)'))</GeneratedOutput>
+        <GeneratedOutput>$(RazorGenerateIntermediateOutputPath)%(RazorGenerateWithTargetPath.TargetPath)$(RazorGenerateOutputFileExtension)</GeneratedOutput>
```

Comment:
Changes from `obj/Debug/netcoreapp3.0/Razor/Index.g.cshtml.cs` ->`obj/Debug/netcoreapp3.0/Razor/Index.cshtml.g.cs`.

Makes it trivial to do `obj/Debug/netcoreapp3.0/RazorComponents/Counter.razor.g.cs`